File: algorithm/leetcode739_daily-temperatures.cpp
class Solution {
public:
    vector<int> dailyTemperatures(vector<int>& T) {
        vector<int> ans(T.size(),0);
        stack<int> stk;
        for(int i = 0; i < T.size(); ++i)
        {
        	while(!stk.empty() && T[stk.top()] < T[i])
        	{
        		ans[i] = i-stk.top();
        		stk.pop();
        	}
        	stk.push(i);
        }
        return ans;
    }
};

File: algorithm/Python3_version/leetcode739.daily-temperatures.py
class Solution:
    def dailyTemperatures(self, T: List[int]) -> List[int]:
        n = len(T)
        ans = [0]*n
        s = []
        for i in range(n-1,-1,-1):
            while len(s)>0 and T[i] >= T[s[-1]]:
                s.pop()
            if len(s)>0:
                ans[i] = s[-1]-i
            s.append(i)
        return ans